include/roboptim/core/function/polynomial.hh

```diff
@@ -72,6 +72,16 @@ namespace roboptim
     value_type applyPolynomial
     (const_vector_ref coeffs, const_argument_ref x) const;
 
+  public:
+    /// \brief Get the type-checking flag
+    virtual unsigned long flag()
+    {
+      return flag_;
+    }
+
+    /// \brief Flag representing the Roboptim Function type
+    static const unsigned long flag_;
+
```

Comment:
Yes, if we have a lower-level function calling `asType` on a higher-level function, everything is fine since it can access `flag_`, but the contrary is not possible, hence the need for a public access. Gotcha. So all is left to do is precise whether we call the static member `flag` or `Flag` (and the plural form would make more sense since it contains all of the flags describing the function), and same for the getter. To be consistent with other parts of the code, e.g.:

```cpp
    static const size_type derivabilityOrder = 2;
```

I guess we can have:

```cpp
    virtual flag_t getFlags()
    {
      return flags;
    }

    static const flag_t flags = ...;
```